components/quick_input/index.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+
+import {setSelectionRange} from 'utils/utils';
+
+import QuickInputTextbox from './quick_input_textbox';
+import ClearableTooltip from './clearable_tooltip';
+import {useRefHandler} from './use_ref_handler';
+import {QuickInputState} from './quick_input_state';
+
+interface IQuickInputProps{
+    delayInputUpdate?: boolean;
+
+    /**
+     * An optional React component that will be used instead of an HTML input when rendering
+     */
+    inputComponent?: any ;
+
+    /**
+     * The string value displayed in this input
+     */
+    value: string;
+
+    /**
+     * When true, and an onClear callback is defined, show an X on the input field that clears
+     * the input when clicked.
+     */
+    clearable?: boolean;
+
+    /**
+     * The optional tooltip text to display on the X shown when clearable. Pass a components
+     * such as FormattedMessage to localize.
+     */
+    clearableTooltipText?: string | JSX.Element;
+
+    /**om
+     * Callback to clear the input value, and used in tandem with the clearable prop above.
+     */
+    onClear?: () => void;
+
+    /**
+     * Input event callback, for old API(using ref)
+     */
+    onChange?: (e: React.ChangeEvent<HTMLInputElement | any>) => void;
+    onInput?: (e: React.ChangeEvent<HTMLInputElement | any>) => void;
+    onCompositionUpdate?: (e: React.CompositionEvent<HTMLInputElement | any>) => void;
+    onSelect?: (e: React.SyntheticEvent<HTMLInputElement, any>) => void;
+    onCompositionStart?: () => void;
+    onCompositionEnd?: () => void;
+    onKeyDown?: (e: KeyboardEvent) => void;
+
+    /**
+     * Input event callback,for new API (using QuickInputState class to control)
+     */
+    quickInputState?: QuickInputState;
+    onTextboxChange?: (qInputState: QuickInputState) => void;
+    onTextboxCompositionUpdate?: (
+        qInputState: QuickInputState, composingData: string | null | undefined
+    ) => void;
+    onTextboxSelect?: (qInputState: QuickInputState) => void;
+}
+
+const QuickInput = React.forwardRef((props: IQuickInputProps, ref) => {
+    //reference to quick input text box
```

Comment:
Our convention for /index.tsx files is to connect the component to redux but not render it within

Reply:
Oh ok, this can be solved easily.